### ISSUE TITLE
Ribbon buttons: add "open in window" and/or push on platform to allow launching windows on all platforms

### ISSUE THREAD

**Issue by Zlatkovsky** [OfficeDev/script-lab#402]
From Mourad Louha's excellent blog post (auto-translated to English):  http://www.excel-ticker.de/prototyping-von-microsoft-office-javascript-add-ins-mit-script-lab/

![image](https://cloud.githubusercontent.com/assets/2230453/25648292/35769946-2f7f-11e7-803a-ef1216887d5d.png)

**Comment by Zlatkovsky** [comment]
@saunders77 , what I mean is having a navigation bar at the top, with a link.  Something like this:

You can use clipboard.js to do the clipboard-copying part.

![image](https://cloud.githubusercontent.com/assets/2230453/25682188/15f53892-300c-11e7-8587-e2fffeb71eb0.png)

**Comment by saunders77** [comment]
Got it, thanks!

**Comment by Hongbo-Miao** [comment]
Personally I think just using default browser may provide better experience

- Without modal, we can let users edit the Excel at the same time.
- For most developers, using existing browser also does not need to log in the Stack Overflow again.
- Simple to implement, just open the link.
- Existing browsers have users own experience. For example, if use ad-blocker, no ads will show

![image](https://cloud.githubusercontent.com/assets/3375461/26610213/1425673a-455b-11e7-9de4-04a86931ffeb.png)

**Comment by Zlatkovsky** [comment]
Yep, 100% agree.  The problem is that doing a `window.open` wasn't working on at least some of the platforms (Office Online, IIRC).

I suppose one possible solution is for us to use OfficeHelpers to check the host type, and use the dialog API for Office Online and `window.open` for the others.  @saunders77 , do you remember if we tried that combo?  it feels like it should work, but I'm also surprised we wouldn't have thought of it before...

Thanks!
- Michael Z.

**Comment by WrathOfZombies** [comment]
What's wrong with just having a link whose domain isn't in the AppDomain list. Secondly OfficeHelpers doesn't seem the place for this. It will unnecessarily bloat things up overtime.

**Comment by WrathOfZombies** [comment]
To that effect the second part of the title, to push on platform makes the most sense to me.

**Comment by Zlatkovsky** [comment]
@WrathOfZombies , I don't mean add new code to OfficeHelpers.  I just mean using platform detection (`OfficeHelpers.Utilities.host`).

I agree that pushing on the platform is also good on this, it's just that we need a stopgap measure.

@saunders77 , if the issue with `window.open` is indeed only on Office Online, I wonder if the correct (and simple) solution is simply to have Office.js overwrite (monkey-patch) window.open to show a UI with a user-clicked `<a href="" />` action that would allow the click through past the popup blocker (much as the dialog API does today if you set the non-iframe option...).  Is this a bug you can file and push on the platform on?...

Thanks!

**Comment by Zlatkovsky** [comment]
@saunders77 , ping
